sql/set_var.cc

```diff
@@ -346,7 +346,7 @@ uchar *sys_var::global_var_ptr() {
 
 bool sys_var::check(THD *thd, set_var *var) {
   if ((var->value && do_check(thd, var)) ||
-      (on_check && on_check(this, thd, var))) {
+      (var->type != OPT_PERSIST_ONLY && on_check && on_check(this, thd, var))) {
```

Comment:
So if I understand correctly, the do_check() now does do_def_check() for PERSIST_ONLY and we skip the on_check() for PERSIST_ONLY.  Right? [the core change for sys_vars]

Reply:
In case of PERSIST_ONLY do_check will invoke do_def_check() for plugin vars and will do "default system" check for sys vars. And on_check() will be skipped here.

Reply:
Ok got it